From: Ilya Dryomov <ilya.dryomov@inktank.com>
Date: Fri, 25 Jul 2014 12:44:58 +0400
Subject: [PATCH] ceph: remove redundant memset(0)
Git-commit: 1a295bd8c854de221813d0ba0848031a8985b869
Patch-mainline: v3.17
References: Fate#318586


xattrs array of pointers is allocated with kcalloc() - no need to
memset() it to 0 right after that.

Signed-off-by: Ilya Dryomov <ilya.dryomov@inktank.com>
Reviewed-by: Alex Elder <elder@linaro.org>
Acked-by: NeilBrown <neilb@suse.com>

---
 fs/ceph/xattr.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/ceph/xattr.c
+++ b/fs/ceph/xattr.c
@@ -587,7 +587,7 @@ start:
 		err = -ENOMEM;
 		if (!xattrs)
 			goto bad_lock;
-		memset(xattrs, 0, numattr*sizeof(struct ceph_xattr *));
+
 		for (i = 0; i < numattr; i++) {
 			xattrs[i] = kmalloc(sizeof(struct ceph_inode_xattr),
 					    GFP_NOFS);
